### PR TITLE
Add missing delete / options method to http module

### PR DESCRIPTION
**Pull request by aisk**

### CODE REVIEW

File: vm/http_test.go [hunk 1]
```diff
@@ -37,7 +37,19 @@ func TestHTTPRequest(t *testing.T) {
 
 		res = Net::HTTP.head("http://127.0.0.1:3000/index")
 		res["Content-Length"]
-		`, "15"},
+		`, "16"},
```
Comment (st0012) [inline vm/http_test.go:40]: can we use the exact content instead of the length of the content for comparison?

**Review by st0012 — COMMENTED** [review]
great work, just a small comment 👍

File: vm/http_test.go (hunk 1 above)
Comment (aisk) [reply vm/http_test.go:40]: Currently head / delete / options only returns HTTP headers. I have post a thought on slack.

Comment (st0012) [reply vm/http_test.go:40]: ah of course, my fault 🤦 
can we just test the status code then (should all be `204`)

Comment (aisk) [reply vm/http_test.go]: Sorry late for the reply, just changed it to check the method from response headers.

**Review by st0012 — APPROVED** [review]

**Comment by st0012** [comment]
@aisk sorry that I forgot to click merge after approving it 😬 mergint now

**Pull request merged**